### ISSUE TITLE
Gold Price Prediction using Machine Learning

### ISSUE THREAD

**Issue by keshripritesh** [PRIYANSHU2026/DataSentience-AIML#708]
### 📌 Issue Title:

\[ GSSoC’25 ] Add New Project — Gold Price Prediction using Machine Learning

---

#### 🚀 Project Idea

This project aims to **predict the price of gold** using various **machine learning models**. It includes **data exploration, feature engineering, model training, hyperparameter tuning, and evaluation**. The goal is to build a reliable predictive model that leverages historical data and financial indicators.

The project will serve as a great contribution for beginners and intermediate contributors in **machine learning, data science, and time series forecasting**.

---

#### 📊 Dataset

The dataset used is publicly available on Kaggle:
🔗 [[Gold Price Prediction Dataset](https://www.kaggle.com/datasets/sid321axn/gold-price-prediction-dataset)](https://www.kaggle.com/datasets/sid321axn/gold-price-prediction-dataset)

It contains **historical gold price data** along with related **financial indicators**, making it suitable for supervised learning.

---

#### 📂 Tech Stack

* **Programming Language:** Python
* **Libraries & Tools:**

  * Data Analysis: `pandas`, `numpy`
  * Visualization: `matplotlib`, `seaborn`
  * Machine Learning: `scikit-learn`, `xgboost`
  * Model Optimization: `GridSearchCV`

---

#### 🔍 Key Features

1. **Exploratory Data Analysis (EDA):**

   * Cleaning missing values
   * Descriptive statistics
   * Visualizations (trends, seasonality, correlations)

2. **Feature Engineering:**

   * Lag features for time-dependency
   * Rolling averages & other statistical features

3. **Model Training & Evaluation:**

   * Models: Random Forest, XGBoost, Decision Trees, Linear Regression, Gradient Boosting
   * Evaluation Metrics: **MAE, MSE, R²**

4. **Hyperparameter Tuning:**

   * Grid Search for best performance

5. **Best Performing Model:**

   * `XGBoost Regressor` with the lowest error and highest accuracy

---

#### ✅ Expected Outcomes

* A **well-documented machine learning project** for gold price prediction
* Contributors gain hands-on experience with:

  * **Time series forecasting**
  * **Feature engineering**
  * **Model selection & optimization**

---

#### 📸 Output Snapshots

* EDA visualizations (correlation heatmaps, trends)

<img width="632" height="515" alt="Image" src="https://github.com/user-attachments/assets/2f949dcf-8883-4c6f-942d-bbec3c7afa89" />

<img width="632" height="515" alt="Image" src="https://github.com/user-attachments/assets/9f1493cb-b813-4b96-ace3-4986fe2abc60" />

* Final results highlighting **XGBoost Regressor**

<img width="1629" height="482" alt="Image" src="https://github.com/user-attachments/assets/5dd6f096-e627-4315-9086-06e90d75060f" />

---
